netbox/extras/webhooks_worker.py

```diff
@@ -33,10 +41,22 @@ def process_webhook(webhook, data, model_name, event, timestamp, username, reque
         'headers': headers
     }
 
-    if webhook.http_content_type == WebhookContentTypeChoices.CONTENTTYPE_JSON:
-        params.update({'data': json.dumps(payload, cls=JSONEncoder)})
-    elif webhook.http_content_type == WebhookContentTypeChoices.CONTENTTYPE_FORMDATA:
-        params.update({'data': payload})
+    logger.info(
+        "Sending webhook to {}: {} {}".format(params['url'], context['model'], context['event'])
+    )
+
+    # Construct the request body. If a template has been defined, use it. Otherwise, dump the context as either JSON
+    # or form data.
+    if webhook.body_template:
+        try:
+            params['data'] = render_jinja2(webhook.body_template, context)
+        except TemplateError as e:
+            logger.error("Error rendering request body: {}".format(e))
+            return
+    elif webhook.http_content_type == HTTP_CONTENT_TYPE_JSON:
```

Comment:
I wonder if we should add an additional sanity check here to catch invalid JSON from a templated body. Basically check if the content type is `HTTP_CONTENT_TYPE_JSON` and a jinja2 rendered body, attempt a json.loads() and log a helpful message on failure. I imagine this would be very helpful to users who are writing JSON templates (because they absolutely will).

Reply:
An opportunistic validation of the outgoing JSON? We can, I guess. But are you proposing that an outgoing request with invalid JSON data gets aborted or just logged as a warning?